File: in_Java/1413 Minimum Value to Get Positive Step by Step Sum.java
class Solution {
    public int minStartValue(int[] nums) {
        int[] cumu = new int[nums.length];
        cumu[0] = nums[0];
        int curmin = nums[0];
        for(int i = 1; i < nums.length; i++){
            cumu[i] = cumu[i-1] + nums[i];
            curmin = Math.min(curmin, cumu[i]);
        }
        return (1 - curmin) < 1? 1 : (1-curmin);
    }
}

File: in_Python/1413 Minimum Value to Get Positive Step by Step Sum.py
class Solution(object):
    def minStartValue(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mm = nums[0]
        cursum = nums[0]
        for i in range(1, len(nums)):
            cursum += nums[i]
            mm = min(mm, cursum)
        #print(mm)
        return abs(min(0, mm)) + 1
        